src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_cpu_blocking.cpp

```diff
@@ -88,47 +63,32 @@ bool BrgemmCPUBlocking::mark_blocking_loops(LinearIR& linear_ir,
         const ov::snippets::VectorDims compensations_subtensor{1, n_block};
         OPENVINO_ASSERT(brgemm_expr->get_input_count() == 3, "Brgemm must have 3 inputs in case of compensations.");
         brgemm_expr->get_input_port_descriptor(2)->set_subtensor(compensations_subtensor);
         copy_b_expr->get_output_port_descriptor(1)->set_subtensor(compensations_subtensor);
     }
+    if (with_amx(type)) {
+        move_new_memory_buffer(linear_ir, brgemm_it);
+        auto buffer_it = std::prev(brgemm_it);
+        buffer_it->get()->set_loop_ids(brgemm_expr->get_loop_ids());
+    }
 
     const auto& loop_manager = linear_ir.get_loop_manager();
-    if (!is_full_dim_value(k_block)) {
-        const auto loop_begin = get_loop_begin_pos(linear_ir, brgemm_it, copy_b_expr);
-        const std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), true, 0),
-                                            LoopPort(copy_b_expr->get_input_port(0), true, 1)};
-        const std::vector<LoopPort> exits{LoopPort(brgemm_expr->get_output_port(0), false)};
-        mark_k_blocking(loop_manager, loop_begin, std::next(brgemm_it), entries, exits, k_block);
-    }
-    if (!is_full_dim_value(n_block)) {
-        const auto loop_begin = get_loop_begin_pos(linear_ir, brgemm_it, copy_b_expr);
-        const std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), false),
-                                            LoopPort(copy_b_expr->get_input_port(0), true)};
-        const std::vector<LoopPort> exits{LoopPort(brgemm_expr->get_output_port(0), true)};
-        mark_n_blocking(loop_manager, loop_begin, std::next(brgemm_it), entries, exits, n_block);
-    }
-    if (!is_full_dim_value(m_block)) {
-        const bool include_repacking = !is_full_dim_value(k_block) || !is_full_dim_value(n_block);
-        const auto loop_begin = get_loop_begin_pos(linear_ir, brgemm_it, include_repacking ? copy_b_expr : nullptr);
-        const auto b_input_port = include_repacking ? copy_b_expr->get_input_port(0) : brgemm_expr->get_input_port(1);
-        std::vector<LoopPort> entries{LoopPort(brgemm_expr->get_input_port(0), true), LoopPort(b_input_port, false)};
-        if (!include_repacking && with_compensations(type))
-            entries.emplace_back(brgemm_expr->get_input_port(2), false);
-        const std::vector<LoopPort> exits{LoopPort(brgemm_expr->get_output_port(0), true)};
-        mark_m_blocking(loop_manager, loop_begin, std::next(brgemm_it), entries, exits, m_block);
+    if (!is_full_dim_value(m_block) && with_compensations(type)) {
+        const LoopPort default_port(brgemm_expr->get_input_port(1), false, 1);
+        const std::vector<LoopPort> replacement_ports {default_port, LoopPort(brgemm_expr->get_input_port(2), false, 1)};
+        auto m_loop_id = brgemm_expr->get_loop_ids().front();
+        loop_manager->get_loop_info<UnifiedLoopInfo>(m_loop_id)->replace_with_new_ports(default_port, replacement_ports);
```

Comment:
That's really cool refactoring of the CPU blocking pass 👍 : now the main logic is placed in base class, and here we have only light-weight corrections

Reply:
BTW shouldn't we update KN loops with new port as well in case of compensations?

Reply:
> BTW shouldn't we update KN loops with new port as well in case of compensations?
We can do it (it shouldn't really change anything), but what for?
The main purpose of adding a non-incrementing loop port is to keep the register alive, so I believe it's enough to update only the outermost loop.
Please correct me if I'm missing smth here.

Reply:
actually, let me think about it for a bit 🤔

Reply:
After the offline discussion, I added the compensation port to all blocking loops. Please take a look.